comp/dogstatsd/server/server.go

```diff
@@ -511,11 +512,54 @@ func (s *server) SetExtraTags(tags []string) {
 // SetBlocklist updates the metric names blocklist on all running worker.
 func (s *server) SetBlocklist(metricNames []string, matchPrefix bool) {
 	s.log.Debugf("SetBlocklist with %d metrics", len(metricNames))
-	// each worker receives its own copy
+
+	// we will use two different blocklists:
+	// - one with all the metrics names, with all values from `metricNames`
+	// - one with only the metric names ending with histogram aggregates suffixes
+
+	// only histogram metric names (including their aggregates suffixes)
+	histoMetricNames := s.createHistogramsBlocklist(metricNames)
+
+	// send the complete blocklist to all workers, the listening part of dogstatsd
 	for _, worker := range s.workers {
-		blocklist := newBlocklist(metricNames, matchPrefix)
+		blocklist := utilstrings.NewBlocklist(metricNames, matchPrefix)
 		worker.BlocklistUpdate <- blocklist
 	}
+
+	// send the histogram blocklist used right before flushing to the serializer
+	histoBlocklist := utilstrings.NewBlocklist(histoMetricNames, matchPrefix)
+	s.demultiplexer.ReconfigTimeSamplersBlocklist(&histoBlocklist)
+}
+
+// create a list based on all `metricNames` but only containing metric names
+// with histogram aggregates suffixes.
+func (s *server) createHistogramsBlocklist(metricNames []string) []string {
+	aggrs := s.config.GetStringSlice("histogram_aggregates")
+
+	percentiles := metrics.ParsePercentiles(s.config.GetStringSlice("histogram_percentiles"))
```

Comment:
Would it make sense to move this function to the `metrics` package? So that all handling of percentiles is concentrated in one, rather than multiple packages.

Reply:
It's use-case is IMO closer to statsd than the percentiles/histograms themselves, don't you think?